### PR TITLE
CORE-19447: Read managed key rotation status

### PR DESCRIPTION
**Pull request by anien**
- Managed key rotation status is now presented similarly to unmanaged key rotation status. Wrapping key alias is shown in the status, as we already show this value when we list key pairs for particular tenant (using existing rest call). 
- When starting managed key rotation, we check if any key rotation is in progress.

### CODE REVIEW

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt [hunk 1]
```diff
@@ -251,11 +273,47 @@ class KeyRotationRestResourceImpl @Activate constructor(
...
+    private fun prepareUnmanagedRotationOutput(records: Collection<State>): List<Pair<String, RotatedKeysStatus>> {
+        return records.map { state ->
+            val unmanagedKeyRotationStatus = checkNotNull(unmanagedKeyStatusDeserializer.deserialize(state.value))
+            unmanagedKeyRotationStatus.tenantId to RotatedKeysStatus(
+                unmanagedKeyRotationStatus.total,
+                unmanagedKeyRotationStatus.rotatedKeys
+            )
+        }
+    }
+
```
Comment (simon-johnson-r3) [inline components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt:295]: I wont block the PR on this, but a slightly more idiomatic Kotlin way to do this (both these methods) which we use in Corda quite frequently is to use an extension function, e.g.:
```
private fun Collection<State>.toUnmanagedRotationOutput() =
    map { state ->
        val unmanagedKeyRotationStatus =
            checkNotNull(unmanagedKeyStatusDeserializer.deserialize(state.value))
        unmanagedKeyRotationStatus.tenantId to RotatedKeysStatus(
                unmanagedKeyRotationStatus.total,
                unmanagedKeyRotationStatus.rotatedKeys
            )
        }
```

**Review by simon-johnson-r3 — DISMISSED** [review]
Approved, one minor possible improvement in code style, but please consider it optional.

File: components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt (hunk 1 above)
Comment (anien) [reply components/crypto/crypto-rest/src/main/kotlin/net/corda/crypto/rest/impl/KeyRotationRestResourceImpl.kt:295]: Changed.

**Review by simon-johnson-r3 — APPROVED** [review]

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**